src/bin/vip-wp.js

```diff
@@ -72,6 +112,66 @@ commandWrapper( {
 
 		let result;
 		let rl;
+		let subShellRl;
+
+		if ( isSubShell ) {
+			console.log( `Entering WP-CLI subshell mode for ${ formatEnvironment( envName ) } on ${ appName } (${ appId })` );
+
+			subShellRl = readline.createInterface( {
+				input: process.stdin,
+				output: process.stdout,
+				terminal: true,
+				prompt: `${ appName }.${ envName }> `,
+				// TODO make history persistent across sessions for same env
+				historySize: 200,
+			} );
+
+			subShellRl.on( 'line', async line => {
+				console.log( 'received line:', line );
+				subShellRl.pause();
+
+				try {
+					result = await getTokenForCommand( appId, envId, line.replace( 'wp ', '' ) );
+				} catch ( e ) {
+					console.log( e );
+
+					return;
+				}
+
+				const { data: { triggerWPCLICommandOnAppEnvironment: { command: cliCommand, inputToken } } } = result;
+
+				const commandStreams = await launchCommandAndGetStreams( {
+					guid: cliCommand.guid,
+					inputToken: inputToken,
+				} );
+
+				commandStreams.stdoutStream.pipe( process.stdout );
+
+				commandStreams.stdoutStream.on( 'error', err => {
+					// TODO handle this better
+					console.log( err );
+
+					process.exit( 1 );
```

Comment:
We probably want to just print the error to the subshell, but not exit, so that the user can run another command after this without starting up the subshell again.

Reply:
Yes indeed 👍 https://github.com/Automattic/vip/pull/280/commits/a009aca16e0a40183fafcc62e2cf8dd6c18b8745